util/prefix_validate.go

```diff
@@ -0,0 +1,35 @@
+package util
+
+import (
+	"errors"
+	"regexp"
+)
+
+/*
+IsValidTagPrefix runs a series of regex checks to ensure that the tag prefix is valid
+  Tags cannot begin or end with, or contain multiple consecutive / characters.
+  They cannot contain any of the following characters \, ?, ~, ^, :, * , [, @.
+  They cannot contain a space.
+  They cannot end with a . or have two consecutive .. anywhere within them.
+  Tags are not case-sensitive.
+*/
+func IsValidTagPrefix(prefix string) (bool, error) {
+	if prefix == "" {
+		return true, nil
+	}
+
+	if !regexp.MustCompile(`^[^/][a-zA-Z\d\.\-_/]*$`).MatchString(prefix) {
+		return false, errors.New("tag prefix contains invalid character(s) or begins with /")
+	}
+	if regexp.MustCompile(`/{2,}`).MatchString(prefix) {
+		return false, errors.New("tag prefix contains multiple consecutive / characters")
+	}
+	if regexp.MustCompile(`\.{2,}`).MatchString(prefix) {
+		return false, errors.New("tag prefix contains multiple consecutive . characters")
+	}
+	if !regexp.MustCompile(`\D$`).MatchString(prefix) {
+		return false, errors.New("tag prefix ends with a digit")
+	}
```

Comment:
Can you please move the regexs to package level variables, and call `MustCompile` in the `init` function? For now, it doesn't make a difference, but if we ever call this function twice, we save time on executing the regexs, as we don't have to recompile them over and over again.

Besides with (this is something you could not know, sorry for that): Can you rephrase the error messages to use the `must` form, e.g. `version prefix must not contain …`?

Oh, and it should be `version prefix`, not `tag prefix` 😉